internal/booking/timeedit/bookingService.go

```diff
@@ -11,26 +11,40 @@ import (
 	"strings"
 	"time"
 
-	"github.com/PuerkitoBio/goquery"
 	"golang.org/x/net/publicsuffix"
 
+	"github.com/PuerkitoBio/goquery"
+
 	"sidus.io/boogrocha/internal/booking"
 )
 
-const samelURL = "https://cloud.timeedit.net/chalmers_test/web/timeedit/sso/saml2_test?back=https%3A%2F%2Fcloud.timeedit.net%2Fchalmers_test%2Fweb%2Fb1%2F"
-const bookURL = "https://cloud.timeedit.net/chalmers_test/web/b1/ri1Q5008.html"
-const objectsURL = "https://cloud.timeedit.net/chalmers_test/web/b1/objects.json?part=t&types=186&step=1"
-const bookingsURL = "https://cloud.timeedit.net/chalmers_test/web/b1/my.html"
+const ChalmersTest = "chalmers_test"
+const Chalmers = "chalmers"
+
+const samlURLFormat = "https://cloud.timeedit.net/%s/web/timeedit/sso/%s?back=https://cloud.timeedit.net/%s/web/b1/"
+const bookURLFormat = "https://cloud.timeedit.net/%s/web/b1/ri1Q5008.html"
+const objectsURLFormat = "https://cloud.timeedit.net/%s/web/b1/objects.json?part=t&types=186&step=1"
+const bookingsURLFormat = "https://cloud.timeedit.net/%s/web/b1/my.html"
 const roomInfoURL = "https://boogrocha.sidus.io/rooms.json"
 const otherPurpose = "203460.192"
-const providerName = "ChalmersTimeEdit"
+const BaseProvider = "TimeEdit"
+
+var studentUnionRooms = []string{
+	"Grupprum 1",
+	"Grupprum 2",
+	"Grupprum 3",
+	"Musikrummet",
+	"Motionshallen",
+}
 
 type BookingService struct {
 	client *http.Client
-	rooms  rooms
+	rooms  Rooms
 }
 
-func (bs BookingService) Book(booking booking.Booking) error {
+func (bs BookingService) Book(booking booking.Booking, timeEditVersion string) error {
```

Comment:
Wouldn't it be more nice to have `timeEditVersion string` as a parameter to the constructor? Would save you a lot of complexity and code